src/main/java/soolutions/hometaskapp/account/Transfer.java

```diff
@@ -0,0 +1,37 @@
+package soolutions.hometaskapp.account;
+
+import soolutions.hometaskapp.account.Account;
+import soolutions.hometaskapp.account.Deposit;
+import soolutions.hometaskapp.account.Transaction;
+import soolutions.hometaskapp.account.Withdrawal;
+import soolutions.hometaskapp.common.Amount;
+
+public final class Transfer {
+    public void apply(Account from, Account to, Amount amount) {
+        precondition(from, to, amount);
+        from.apply(new Withdrawal(amount, "Transfer to " + to));
+        to.apply(new Deposit(amount, "Transfer from " + from));
+    }
+
+    private void precondition(Account from, Account to, Amount amount) {
+        if (!from.open()) {
+          throwException("Given account is closed " + from);
+        }
+        if (!to.open()) {
+          throwException("Given account is closed " + to);
+        }
+        if (amount.compareTo(from.balance()) < 0) {
+          throwException("Source account balance cannot be negative " + from);
```

Comment:
Is this a correct exception message, because it looks like it says: "Hey, right now your source accout has negative balance, think about this", but what, I think, it could say: "Hey, you don't have enough money to transfer such amount".

So it can be like:
`throwException( "Source account doesn't have enough balance: transfering" + amount + " but has " + from.balance() + ".");`.